src/seedu/addressbook/data/person/Address.java

```diff
@@ -25,34 +36,111 @@ public Address(String address, boolean isPrivate) throws IllegalValueException {
         if (!isValidAddress(address)) {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = address;
+//        this.value = address;
+        assignAddressComponents(address);
+    }
+    
+    /**
+     * Assign the address components into its individual classes.
+     * Only takes in the following address format a/BLOCK, STREET, UNIT, POSTAL_CODE
+     *
+     * @param string containing the entire address
+     */
+    private void assignAddressComponents(String address) throws IllegalValueException {
+    	final String[] addressComponents = address.split(ADDRESS_DELIMITER);
```

Comment:
Rather than splitting by comma and a space, consider split by comma then trim the space which is more user-friendly. Since user might often forget to put a space after comma.